src/ui/component/field/numericField.tsx

```diff
@@ -0,0 +1,52 @@
+import type { FC, ChangeEvent } from 'react'
+import { useCallback, useState } from 'react'
+import classNames from 'classnames'
+import type { NumericFormatProps } from 'react-number-format'
+import { NumericFormat } from 'react-number-format'
+import type { BaseFieldProps } from './baseField'
+import { BaseField } from './baseField'
+import './field.scss'
+
+type NumericFieldProps = NumericFormatProps & BaseFieldProps
+
+export const NumericField: FC<NumericFieldProps> = ({
+  label,
+  error,
+  leftElement,
+  rightElement,
+  ...inputProps
+}) => {
+  const { id, onChange } = inputProps
+
+  const [focused, setFocused] = useState(false)
+  const handleFocus = useCallback((): void => setFocused(true), [])
+  const handleBlur = useCallback((): void => setFocused(false), [])
```

Comment:
I have a question about the way of handling blur and focus: I can see there is a classname which depends on the focus state, but in `onBlur={handleBlur}`, you don't passe the `onBlur` method given by the user.
You should add it into `handleBlur` and same for `onFocus`, what do you think?
And also, line from 21 to 23 should directly get into baseField no? same for the focused className, and it avoir you to repeat same logic for numerical and textField components.
Not sure about all this but I think it can be improved !

Reply:
You are definitely right, `onBlur` and `onFocus` are used here to manage field style. They don't need to be used in both Numeric and Text fields.

In order to simplify the component, onBlur and onFocus event handlers has been removed and styling has been done in CSS. Input tag has been moved as the first child of field main div to apply CSS style using `~` sibling operator, until the `:has` keyword is supported on Firefox browser.

Fields components are simplified, details of the change was done in [a862e5a](https://github.com/okp4/dataverse-portal/pull/430/commits/a862e5af1f51be331c5deeeb023dbb9572d7c71b).